sql: decode all plan gists in test builds

yuzefovich opened the pull request:
**explain: remove a couple redundant nil checks**

`OutputBuilder.Expr` already checks for `nil` argument, so we can skip doing that in a few places.

**sql: decode all plan gists in test builds**

This commit adds the logic to always decode plan gists in test builds right after the gist was created. This should have trivial overhead while providing extra test coverage for the feature.

Note that we had `TestExplainGist` that was targeting this coverage, so to a certain degree that test is now obsolete. However, I decided to not completely remove it since the test has been good at finding issues unrelated to gists (it is effectively the `sqlsmith` roachtest that runs as a unit test).

Informs: #143211
Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143251)
<!-- Reviewable:end -->

yuzefovich committed:
explain: remove a couple redundant nil checks

`OutputBuilder.Expr` already checks for `nil` argument, so we can skip
doing that in a few places.

Release note: None

yuzefovich committed:
sql: decode all plan gists in test builds

This commit adds the logic to always decode plan gists in test builds
right after the gist was created. This should have trivial overhead
while providing extra test coverage for the feature.

Note that we had `TestExplainGist` that was targeting this coverage, so
to a certain degree that test is now obsolete. However, I decided to not
completely remove it since the test has been good at finding issues
unrelated to gists (it is effectively the `sqlsmith` roachtest that runs
as a unit test).

Release note: None

yuzefovich commented:
Friendly ping @cockroachdb/sql-queries-prs

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/143251#-:-OP68Ar6Evfk7kYH7B1u:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
